### PR TITLE
GEODE-7864: Fix potential resource leak LGTM warnings

### PR DESCRIPTION
**Pull request by DonalEvans**
Authored-by: Donal Evans <doevans@vmware.com>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/DeployCommand.java [hunk 1]
```diff
@@ -101,26 +101,36 @@ public ResultModel deploy(
...
+      try {
+        for (String jarFullPath : jarFullPaths) {
+          FileInputStream fileInputStream = null;
+          try {
+            fileInputStream = new FileInputStream(jarFullPath);
+            remoteStreams.add(exporter.export(new SimpleRemoteInputStream(fileInputStream)));
+            jarNames.add(FilenameUtils.getName(jarFullPath));
+          } catch (Exception ex) {
+            if (fileInputStream != null) {
+              fileInputStream.close();
```
Comment (upthewaterspout) [inline geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/DeployCommand.java:113]: Does this need to swallow IOExceptions?

File: geode-tcp-server/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpClient.java [hunk 2]
```diff
@@ -200,21 +200,28 @@ public Object requestToServer(HostAndPort addr, Object request, int timeout,
       out.flush();
 
       if (replyExpected) {
-        DataInputStream in = new DataInputStream(sock.getInputStream());
-        if (debugVersionMessage != null && logger.isDebugEnabled()) {
-          logger.debug(debugVersionMessage);
-        }
-        in = new VersionedDataInputStream(in, serverVersion);
+        DataInputStream in = null;
         try {
```
Comment (upthewaterspout) [inline geode-tcp-server/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpClient.java:204]: Could this have used a `try-with-resources`?

**Review by upthewaterspout — APPROVED** [review]
Looks good! Just a couple comments, inline.

For any other reviewers - it helps to use the github setting to ignore whitespaces changes.

File: geode-tcp-server/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpClient.java (hunk 2 above)
Comment (DonalEvans) [reply geode-tcp-server/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpClient.java]: Good catch, thanks.

File: geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/DeployCommand.java (hunk 1 above)
Comment (DonalEvans) [reply geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/DeployCommand.java]: Fixed

**Review by alb3rtobr — APPROVED** [review]

**Review by rhoughton-pivot — APPROVED** [review]

**Pull request merged**